core/src/test/kotlin/com/exactpro/th2/read/db/impl/grpc/DataBaseReaderGrpcServerIntegrationTest.kt

```diff
@@ -0,0 +1,377 @@
+import java.time.Instant
+import java.time.LocalDate
+import java.time.ZoneOffset
+import java.time.temporal.ChronoUnit
+import java.util.concurrent.TimeUnit
+import kotlin.test.assertNotNull
+
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@IntegrationTest
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class DataBaseReaderGrpcServerIntegrationTest {
+    private val mysql = MySqlContainer()
+    private val persons = (1..30).map {
+        Person("person$it", Instant.now().truncatedTo(ChronoUnit.DAYS), "test-data-$it".toByteArray())
+    }
+
+    private data class Person(val name: String, val birthday: Instant, val data: ByteArray) {
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (javaClass != other?.javaClass) return false
+
+            other as Person
+
+            if (name != other.name) return false
+            if (birthday != other.birthday) return false
+            if (!data.contentEquals(other.data)) return false
+
+            return true
+        }
+
+        override fun hashCode(): Int {
+            var result = name.hashCode()
+            result = 31 * result + birthday.hashCode()
+            result = 31 * result + data.contentHashCode()
+            return result
+        }
+    }
+
+    @BeforeAll
+    fun init() {
+        mysql.withDatabaseName("test_data").start()
+    }
+
+    @AfterAll
+    fun cleanup() {
+        mysql.stop()
+    }
+
+    @BeforeEach
+    fun setup() {
+        execute {
+            dropTable()
+            initTestData()
+        }
+    }
+
+    @Test
+    fun `test execute gRPC request`() {
+        val genericUpdateListener = mock<UpdateListener> { }
+        val genericRowListener = mock<RowListener> { }
+        val messageLoader = mock<MessageLoader> { }
+        val onEvent = mock<OnEvent> { }
+        val streamObserver = mock<StreamObserver<QueryResponse>> { }
+
+        val sourceId = "persons"
+        val queryId = "select"
+        val cfg = DataBaseReaderConfiguration(
+            mapOf(
+                DataSourceId(sourceId) to DataSourceConfiguration(
+                    mysql.jdbcUrl,
+                    mysql.username,
+                    mysql.password,
+                )
+            ),
+            mapOf(
+                QueryId(queryId) to QueryConfiguration(
+                    "SELECT * FROM test_data.person WHERE \${fromId:integer} < id AND id < \${toId:integer}",
+                    mapOf(
+                        "fromId" to listOf(Int.MIN_VALUE.toString()),
+                        "toId" to listOf(Int.MAX_VALUE.toString())
+                    )
+                ),
+            )
+        )
+        val request = QueryRequest.newBuilder().apply {
+            sourceIdBuilder.setId(sourceId)
+            queryIdBuilder.setId(queryId)
+            parametersBuilder
+                .putValues("fromId", "10")
+                .putValues("toId", "20")
+        }.build()
+
+        runTest {
```

Comment:
I don't think you need to use a test scope here. We are testing the gRPC. So, we can simply use `runBlocking` here instead. It will allow us to use blocking calls for gRPC instead of mocking an observer